File: frontend/src/api/audit.ts
import { http } from './http';
import type { AuditLog, AuditLogQuery, PageResult } from './types';

/** 后端仅支持 username/cluster/action 过滤 + 分页(无时间范围) */
export function listAuditLogs(query: AuditLogQuery): Promise<PageResult<AuditLog>> {
  return http.get<PageResult<AuditLog>>('/audit-logs', {
    username: query.username,
    cluster: query.cluster,
    action: query.action,
    page: query.page,
    size: query.size,
    sortBy: query.sortBy,
    order: query.order,
  });
}

File: backend/internal/api/handler/audit.go
package handler

import (
	"github.com/gin-gonic/gin"

	"github.com/v911/backend/internal/api/response"
	"github.com/v911/backend/internal/pkg/errcode"
	"github.com/v911/backend/internal/pkg/pagination"
	"github.com/v911/backend/internal/service"
)

// GET /api/v1/audit-logs(管理员;审计记录只读,无删除接口)
func listAuditLogs(d Deps) gin.HandlerFunc {
	return func(c *gin.Context) {
		p, err := pagination.Parse(c.Request.URL.Query())
		if err != nil {
			response.Fail(c, errcode.New(errcode.ParamInvalid, "invalid pagination").WithCause(err))
			return
		}
		page, err := d.Audit.List(c.Request.Context(), service.AuditQuery{
			Username: c.Query("username"),
			Cluster:  c.Query("cluster"),
			Action:   c.Query("action"),
		}, p)
		if err != nil {
			response.Fail(c, err)
			return
		}
		response.OK(c, page)
	}
}
